mungegithub/submit-queue/www/script.js

```diff
@@ -61,25 +61,43 @@ function SQCntl(dataService, $interval, $location) {
   // Defer loading of data for a tab until it's actually needed.
   // This speeds up the first-boot experience a LOT.
   function loadTab() {
-    // tab: [reloading-function, update interval in minutes, ...] (can repeat)
+    // reloadFunctions is a map from functions to how often they should run (in minutes)
+    var reloadFunctions = {}
+    reloadFunctions[refreshPRs] = 10;
+    reloadFunctions[refreshGithubE2E] = 0.5;
+    reloadFunctions[refreshSQStats] = 30;
+    reloadFunctions[refreshHistoryPRs] = 1;
+    reloadFunctions[refreshE2EHealth] = 10;
+    reloadFunctions[refreshUsers] = 15;
+    reloadFunctions[refreshBotStats] = 10;
+
+    // tabFunctionReloads is a map of which tabs need which functions to refresh
     var tabFunctionReloads = {
-      0: [refreshPRs, 10],
-      1: [refreshGithubE2E, 0.5, refreshSQStats, 30],
-      2: [refreshHistoryPRs, 1],
-      3: [refreshE2EHealth, 10],
-      4: [refreshUsers, 15, refreshBotStats, 10],
+      0: [refreshPRs],
+      1: [refreshGithubE2E, refreshSQStats],
+      2: [refreshHistoryPRs],
+      3: [refreshE2EHealth, refreshSQStats],
+      4: [refreshSQStats, refreshUsers, refreshBotStats],
     }
-    if (self.tabLoaded[self.selected])
+    if (self.tabLoaded[self.selected]) {
       return;
-    var tabReload = tabFunctionReloads[self.selected];
-    if (tabReload !== undefined) {
-      for (var i = 0; i < tabReload.length; i += 2) {
-        var func = tabReload[i];
-        var updateIntervalMinutes = tabReload[i + 1];
-        func();
-        $interval(func, updateIntervalMinutes * 60 * 1000);
+    }
+    self.tabLoaded[self.selected] = true;
+
+    var reloadFuncs = tabFunctionReloads[self.selected];
+    if (reloadFuncs === undefined)
+      return;
+
+    for (var i = 0; i < reloadFuncs.length; i++) {
+      var func = reloadFuncs[i];
+      if (self.functionLoading[func] == true) {
+        continue;
       }
-      self.tabLoaded[self.selected] = true;
+      self.functionLoading[func] = true;
+
+      var updateIntervalMinutes = reloadFunctions[func];
+      func();
```

Comment:
yeah, this calls `func` 'right now' and then the next line calls it 'every so often'